events/src/test/java/com/app/events/controller/SectorCapacityControllerIntegrationTest.java

```diff
@@ -0,0 +1,140 @@
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.annotation.Rollback;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+/**
+ * SectorCapacityControllerIntegrationTest
+ */
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment = WebEnvironment.RANDOM_PORT)
+@TestPropertySource("classpath:application-test.properties")
+public class SectorCapacityControllerIntegrationTest {
+
+    private String authTokenAdmin = "";
+
+	@Autowired
+	private TestRestTemplate restTemplate;
+
+    @Autowired
+    private SectorCapacityRepository sectorCapacityRepository;
+    
+	@Before
+	public void login() throws Exception {
+		LoginDTO loginDto = new LoginDTO(UserConstants.DB_ADMIN_USERNAME, UserConstants.DB_ADMIN_PASSWORD);
+		ResponseEntity<String> response = restTemplate.postForEntity("/api/login", loginDto, String.class);
+		authTokenAdmin = response.getBody();
+    }
+    
+    @Test
+    public void findOneById_when_validID_then_return_SectorCapacityDTO() throws Exception
+    {
+        URI uri = new URI(SectorCapacityConstants.URI_PREFIX + SectorCapacityConstants.PERSISTED_SECTOR_CAPACITY_ID);
+
+		ResponseEntity<SectorCapacityDTO> response = restTemplate.getForEntity(uri, SectorCapacityDTO.class);
+		SectorCapacityDTO found = response.getBody();
+
+        assertEquals(SectorCapacityConstants.PERSISTED_SECTOR_CAPACITY_ID, found.getId());
+        assertEquals(SectorCapacityConstants.PERSISTED_SECTOR_CAPACITY_CAPACITY, found.getCapacity());
+        assertEquals(SectorCapacityConstants.PERSISTED_SECTOR_CAPACITY_FREE, found.getFree());
+        assertEquals(SectorCapacityConstants.PERSISTED_SECTOR_CAPACITY_SECTOR_ID, found.getSector().getId());
+    }
+
+    @Test
+    public void findOneById_when_InvalidID_then_returnNotFound() throws Exception
+    {
+        URI uri = new URI(SectorCapacityConstants.URI_PREFIX + SectorCapacityConstants.INVALID_SECTOR_CAPACITY_ID);
+
+		ResponseEntity<SectorCapacityDTO> response = restTemplate.getForEntity(uri, SectorCapacityDTO.class);
+        SectorCapacityDTO found = response.getBody();
+        
+        assertEquals(null, found.getId());
+        assertEquals(HttpStatus.INTERNAL_SERVER_ERROR, response.getStatusCode());
+    }
+
+
+    @Test
+    @Transactional
+    @Rollback(true)
+    public void create_when_validDataPassedIn_then_addNew() throws Exception
```

Comment:
mislim da nam nikad nece trebati kreiranje sektor capacity, tj da ga pozivamo, da ga imamo u kontroleru, jer se on kreira kad kad se kreira dogadjaj, i molim te skloni proveru da li je null, stavi sectorCapacity.setId(null) i to je to, jedino sto mozes dodati jeste update sector capacity ako cemo to imati da može da prosiri kapacitet ili da smanji, ukoliko smanjuje provera da li postoji mogucnost da smanji tj da li je broj prodatih karata za taj sektor veci od broja na koji zeli da smanji kapacicet